src/main/java/internet/shop/controller/product/DeleteProductController.java

```diff
@@ -0,0 +1,24 @@
+package internet.shop.controller.product;
+
+import internet.shop.lib.Injector;
+import internet.shop.service.ProductService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("internet.shop");
+    private ProductService productService
+            = (ProductService) injector.getInstance(ProductService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String id = req.getParameter("id");
+        Long longId = Long.parseLong(id);
+        productService.delete(longId);
+        resp.sendRedirect(req.getContextPath() + "/admin/products/access");
```

Comment:
let's use the same redirect in AddProductController after we added a new product to the storage